src/material/autocomplete/testing/autocomplete-harness.ts

```diff
@@ -68,28 +69,33 @@ export class MatAutocompleteHarness extends ComponentHarness {
     return (await this.host()).sendKeys(value);
   }
 
-  /** Gets the autocomplete panel. */
-  async getPanel(): Promise<TestElement> {
-    return this._panel();
-  }
-
   /** Gets the options inside the autocomplete panel. */
-  async getOptions(): Promise<MatAutocompleteOptionHarness[]> {
-    return this._options();
+  async getOptions(filters: OptionHarnessFilters = {}): Promise<MatAutocompleteOptionHarness[]> {
+    return this._documentRootLocator.locatorForAll(MatAutocompleteOptionHarness.with(filters))();
   }
 
   /** Gets the groups of options inside the panel. */
-  async getOptionGroups(): Promise<MatAutocompleteOptionGroupHarness[]> {
-    return this._groups();
+  async getOptionGroups(filters: OptionGroupHarnessFilters = {}):
+      Promise<MatAutocompleteOptionGroupHarness[]> {
+    return this._documentRootLocator.locatorForAll(
+        MatAutocompleteOptionGroupHarness.with(filters))();
   }
 
-  /** Gets whether the autocomplete panel is visible. */
-  async isPanelVisible(): Promise<boolean> {
-    return (await this._panel()).hasClass('mat-autocomplete-visible');
+  /** Selects the first options matching the given filters. */
+  async selectOption(filters: OptionHarnessFilters = {}): Promise<void> {
+    if (!await this.isOpen()) {
+      throw Error('mat-autocomplete dropdown must be open to select an option');
+    }
+    const options = await this.getOptions(filters);
+    if (!options.length) {
+      throw Error(`Could not find a mat-option matching ${JSON.stringify(filters)}`);
+    }
+    await options[0].click();
   }
 
   /** Gets whether the autocomplete is open. */
   async isOpen(): Promise<boolean> {
-    return !!(await this._optionalPanel());
+    const panel = await this._optionalPanel();
+    return !!panel && await panel.hasClass('mat-autocomplete-visible');
```

Comment:
One thing to keep in mind here is that doing this will cause `isOpen` to return false if all options were filtered out so you could get into a scenario where you do:

1. Type something -> `isOpen` returns true.
2. Type some more where no options are shown -> `isOpen` returns false.
3. Delete some characters so an option is rendered -> `isOpen` returns true again.

This could end up being confusing for people that don't know how it works.

Reply:
Isn't this how the `isOpen` method on the autocomplete works though? How about if we changed the name to something like `hasVisibleOptions`, would that be more clear?

Reply:
At that point isn't it basically the same as `isPanelVisible`?

Reply:
We could also call it `isPanelVisible` the important thing to me is that there's only one method, not two